atom/browser/native_browser_view_mac.mm

```diff
@@ -8,10 +8,100 @@
 #include "skia/ext/skia_utils_mac.h"
 #include "ui/gfx/geometry/rect.h"
 
+using base::scoped_nsobject;
+
 // Match view::Views behavior where the view sticks to the top-left origin.
 const NSAutoresizingMaskOptions kDefaultAutoResizingMask =
     NSViewMaxXMargin | NSViewMinYMargin;
 
+@interface DragRegionView : NSView
+
+@property (assign) NSPoint initialLocation;
+
+@end
+
+@implementation DragRegionView
+
+- (BOOL)mouseDownCanMoveWindow
+{
+  return NO;
+}
+
+- (NSView *)hitTest:(NSPoint)aPoint
+{
+    // Pass-through events that don't hit one of the exclusion zones
+    for (NSView *exlusion_zones in [self subviews]) {
+      if ([exlusion_zones hitTest:aPoint])
+        return nil;
+    }
+
+    return self;
+}
+
+- (void)mouseDown:(NSEvent *)event
+{
+  if ([self.window respondsToSelector:@selector(performWindowDragWithEvent:)]) {
+    [self.window performWindowDragWithEvent:event];
+    return;
+  }
+
+  self.initialLocation = [event locationInWindow];
+}
+
+- (void)mouseDragged:(NSEvent *)theEvent
+{
+  if ([self.window respondsToSelector:@selector(performWindowDragWithEvent:)]) {
+    return;
+  }
+
+  NSPoint currentLocation;
+  NSPoint newOrigin;
+
+  NSRect screenFrame = [[NSScreen mainScreen] frame];
+  NSRect windowFrame = [self.window frame];
+
+  currentLocation = [NSEvent mouseLocation];
+  newOrigin.x = currentLocation.x - self.initialLocation.x;
+  newOrigin.y = currentLocation.y - self.initialLocation.y;
+
+  // Don't let window get dragged up under the menu bar
+  if( (newOrigin.y + windowFrame.size.height) > (screenFrame.origin.y + screenFrame.size.height) ){
+    newOrigin.y = screenFrame.origin.y + (screenFrame.size.height - windowFrame.size.height);
+  }
+
+  // Move the window to the new location
+  [self.window setFrameOrigin:newOrigin];
+}
+
+// Debugging tips:
+// Uncomment the following four lines to color DragRegionView bright red
```

Comment:
Maybe this could be in a:

```
#ifdef DEBUG_DRAG_REGIONS
...
#endif
```

with a comment at the top:

```
// Uncomment to...
//#define DEBUG_DRAG_REGIONS
```